ufls_detection/ufls_detection_voltage.R

```diff
@@ -0,0 +1,40 @@
+
+ufls_detection_voltage <- function(combined_data, event_time, window_length, 
+                                   fill_nans=FALSE){
+  pre_event_window_start <- event_time - 60 * window_length
+  post_event_window_end <- event_time + 60 * window_length
+  pre_event_window <- filter(combined_data, 
+                             ts >= pre_event_window_start, 
+                             ts < event_time)
+  post_event_window <- filter(combined_data, 
+                              ts > event_time, 
+                              ts <= post_event_window_end)
+
+  pre_event_voltage <- calc_average_voltage_per_circuit(pre_event_window)
+  names(pre_event_voltage)[names(pre_event_voltage) == 'v_mean'] <- 'pre_event_v_mean'
```

Comment:
Definitely my own lack of understanding of R here, but what is the **names(pre_event_voltage) == 'v_mean'** part of this code doing?

Reply:
names(pre_event_voltage) returns a list of the column names within the dataframe 'pre_event_voltage'. The names(pre_event_voltage) == 'v_mean' gives us the index within that list of the name 'v_mean'. Putting that in square brackets means we replace the entry 'v_mean' within the column names with the new column name 'pre_event_v_mean'.

Reply:
Brilliant, thank you!